transport/probe/src/neighbors.rs

```diff
@@ -0,0 +1,132 @@
+use async_stream::stream;
+use futures::Stream;
+#[cfg(all(feature = "prometheus", not(test)))]
+use hopr_metrics::{histogram_start_measure, metrics::SimpleHistogram};
+use libp2p_identity::PeerId;
+use rand::seq::SliceRandom;
+
+use crate::{config::ProbeConfig, store::PeerDiscoveryFetch};
+
+#[cfg(all(feature = "prometheus", not(test)))]
+lazy_static::lazy_static! {
+    static ref METRIC_TIME_TO_HEARTBEAT: SimpleHistogram =
+        SimpleHistogram::new(
+            "hopr_heartbeat_round_time_sec",
+            "Measures total time in seconds it takes to probe all other nodes",
+            vec![0.5, 1.0, 2.5, 5.0, 10.0, 15.0, 30.0, 60.0],
+        ).unwrap();
+}
+
+/// Serializable and deserializable enum for the probe message content
+#[derive(Debug, Clone, Copy, PartialEq, serde::Serialize, serde::Deserialize)]
+pub enum NeighborProbe {
+    /// Ping message with a random nonce
+    Ping(u128),
+    /// Pong message repliying to a specific nonce
+    Pong(u128),
```

Comment:
If the `u128`s are not used for any arithmetic, it would be more expressive to use `[u8; 16]` instead, potentially defining 16 as some ping size constant.

Reply:
They are used to `PartialEq` and `Eq` each other.

Reply:
But `[u8; N]` is `PartialEq` and `Eq` by default.